### PR TITLE
Rebase FFT on kokkos-fft

### LINKED ISSUE #502: Rebase on KokkosFFT

### PR DESCRIPTION
**Pull request by blegouix**
Closes #502

User has responsibility to activate `KokkosFFT_ENABLE_ROCFFT`

### CODE REVIEW

File: include/ddc/kernels/fft.hpp [hunk 1]
```diff
@@ -357,224 +136,127 @@ double b(ddc::DiscreteDomain<DDimX...> x_mesh)
...
+// impl
+template <
+        typename Tin,
+        typename Tout,
+        typename ExecSpace,
+        typename MemorySpace,
+        typename layout_in,
+        typename layout_out,
+        typename DomainOut,
+        typename... DDimIn>
```
Comment (blegouix) [inline include/ddc/kernels/fft.hpp:180]: I would have prefered to keep things symmetric and template with `typename... DDimIn, typename... DDimOut`. But it seems there is a GPU limitation (that could be very annoying when going for a batched FFT):

image

File: include/ddc/periodic_sampling.hpp [hunk 2]
```diff
@@ -302,14 +316,14 @@ template <class DDim, std::enable_if_t<is_periodic_sampling_v<DDim>, int> = 0>
...
-    return coordinate(d.front());
+    return coordinate_without_periodicity(d.front());
 }
 
 template <class DDim, std::enable_if_t<is_periodic_sampling_v<DDim>, int> = 0>
 KOKKOS_FUNCTION Coordinate<typename DDim::continuous_dimension_type> rmax(
         DiscreteDomain<DDim> const& d)
 {
-    return coordinate(d.back());
+    return coordinate_without_periodicity(d.back());
```
Comment (blegouix) [inline include/ddc/periodic_sampling.hpp:326]: Otherwise it can be negative, and `rlength()` (used in FULL normalization) returns wrong result.

**Comment by tpadioleau** [comment]
From yesterday meeting this work has not been identified as a priority

**Comment by blegouix** [comment]
It works when run by hand but stalled.

**Comment by yasahi-hpc** [comment]
What do you mean? Does this work or not?

**Comment by yasahi-hpc** [comment]
Sure. I will have a look

**Comment by tpadioleau** [comment]
Closing because not planned yet, keeping the branch.

**Pull request closed without merging**